### PR TITLE
Update Watchexec & add ARM64 binaries

### PR DESCRIPTION
**Pull request by dmikusa**
## Summary

Attempt at adding in ARM64 support.

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: .github/pipeline-descriptor.yml [hunk 1]
```diff
@@ -27,5 +27,14 @@ dependencies:
     glob:       watchexec-.+-x86_64-unknown-linux-musl.tar.xz
     owner:      watchexec
     repository: watchexec
-    tag_filter: cli-v(1.*)
+    tag_filter: ^v(1\.*)
+    token:      ${{ secrets.PAKETO_BOT_GITHUB_TOKEN }}
+- name:            Watchexec CLI AMD64
```
Comment (anthonydahanne) [inline .github/pipeline-descriptor.yml:32]: ARM64

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml [hunk 2]
```diff
@@ -0,0 +1,118 @@
+name: Update Watchexec CLI AMD64
```
Comment (anthonydahanne) [inline .github/workflows/pb-update-watchexec-cli-amd-64.yml:1]: ARM64; same for title

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                VERSION_PATTERN: '[\d]+\.[\d]+\.[\d]+'
+            - uses: peter-evans/create-pull-request@v5
+              with:
+                author: ${{ secrets.JAVA_GITHUB_USERNAME }} <${{ secrets.JAVA_GITHUB_USERNAME }}@users.noreply.github.com>
+                body: Bumps `Watchexec CLI AMD64` from `${{ steps.buildpack.outputs.old-version }}` to `${{ steps.buildpack.outputs.new-version }}`.
+                branch: update/buildpack/watchexec-cli-amd-64
+                commit-message: |-
+                    Bump Watchexec CLI AMD64 from ${{ steps.buildpack.outputs.old-version }} to ${{ steps.buildpack.outputs.new-version }}
+
+                    Bumps Watchexec CLI AMD64 from ${{ steps.buildpack.outputs.old-version }} to ${{ steps.buildpack.outputs.new-version }}.
```
Comment (anthonydahanne) [inline .github/workflows/pb-update-watchexec-cli-amd-64.yml:113]: duplicate

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                SHA256: ${{ steps.dependency.outputs.sha256 }}
+                SOURCE_SHA256: ${{ steps.dependency.outputs.source_sha256 }}
+                SOURCE_URI: ${{ steps.dependency.outputs.source }}
+                URI: ${{ steps.dependency.outputs.uri }}
+                VERSION: ${{ steps.dependency.outputs.version }}
+                VERSION_PATTERN: '[\d]+\.[\d]+\.[\d]+'
+            - uses: peter-evans/create-pull-request@v5
+              with:
+                author: ${{ secrets.JAVA_GITHUB_USERNAME }} <${{ secrets.JAVA_GITHUB_USERNAME }}@users.noreply.github.com>
+                body: Bumps `Watchexec CLI AMD64` from `${{ steps.buildpack.outputs.old-version }}` to `${{ steps.buildpack.outputs.new-version }}`.
```
Comment (anthonydahanne) [inline .github/workflows/pb-update-watchexec-cli-amd-64.yml:108]: ARM64

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                branch: update/buildpack/watchexec-cli-amd-64
+                commit-message: |-
+                    Bump Watchexec CLI AMD64 from ${{ steps.buildpack.outputs.old-version }} to ${{ steps.buildpack.outputs.new-version }}
+
+                    Bumps Watchexec CLI AMD64 from ${{ steps.buildpack.outputs.old-version }} to ${{ steps.buildpack.outputs.new-version }}.
+                delete-branch: true
+                labels: ${{ steps.buildpack.outputs.version-label }}, type:dependency-upgrade
+                signoff: true
+                title: Bump Watchexec CLI AMD64 from ${{ steps.buildpack.outputs.old-version }} to ${{ steps.buildpack.outputs.new-version }}
```
Comment (anthonydahanne) [inline .github/workflows/pb-update-watchexec-cli-amd-64.yml:117]: ARM64

File: buildpack.toml [hunk 6]
```diff
@@ -36,22 +36,32 @@ api = "0.7"
...
     name = "Watchexec"
     purl = "pkg:generic/watchexec@1.20.5?arch=amd64"
+    source = "https://github.com/watchexec/watchexec/archive/refs/tags/v1.23.0.tar.gz"
+    source-sha256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+    stacks = ["*"]
     sha256 = "33e091870736833b77b103a8e7268c0ae66d756b8d71203405353eb9d40774f2"
-    source = "https://github.com/watchexec/watchexec/archive/refs/tags/cli-v1.20.5.tar.gz"
-    source-sha256 = "2bc04c7ecf58d34a48c3eeea54a76b7e621717cb93305497bea2f6399dd119c6"
-    stacks = ["io.buildpacks.stacks.bionic", "io.paketo.stacks.tiny", "*"]
     uri = "https://github.com/watchexec/watchexec/releases/download/cli-v1.20.5/watchexec-1.20.5-x86_64-unknown-linux-musl.tar.xz"
```
Comment (anthonydahanne) [inline buildpack.toml:43]: weird, should be 1.23.0 too now?

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml (hunk 2 above)
Comment (robdimsdale) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:1]: This workflow is called `amd-64` though, so isn't this correct?

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml (hunk 3 above)
Comment (robdimsdale) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:113]: This workflow is called `amd-64` though, so isn't this correct?

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml (hunk 4 above)
Comment (robdimsdale) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:108]: This workflow is called `amd-64` though, so isn't this correct?

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml (hunk 5 above)
Comment (robdimsdale) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:117]: This workflow is called `amd-64` though, so isn't this correct?

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml (hunk 2 above)
Comment (robdimsdale) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:1]: Oh, I see `glob: watchexec-.+-aarch64-unknown-linux-musl.tar.xz` so I think this _is_ arm, it's just that the workflow file is named `amd64`.

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml (hunk 3 above)
Comment (robdimsdale) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:113]: ~~This workflow is called `amd-64` though, so isn't this correct?~~ see other comment

File: .github/workflows/pb-update-watchexec-cli-amd-64.yml (hunk 2 above)
Comment (anthonydahanne) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:1]: yeah that's right!
1 workflow remains unchanged, the amd64
the other is updating the arm64 dependency

Comment (dmikusa) [reply .github/workflows/pb-update-watchexec-cli-amd-64.yml:1]: Yes, letters were blending together in my head. Sorry for the confusion.

File: buildpack.toml (hunk 6 above)
Comment (dmikusa) [reply buildpack.toml:43]: I just messed that up too. :) Fixing. Good catch.

File: buildpack.toml [hunk 7]
```diff
@@ -32,26 +32,36 @@ api = "0.7"
...
+  [[metadata.dependencies]]
+    cpes = ["cpe:2.3:a:watchexec:watchexec:1.23.0:*:*:*:*:*:*:*"]
+    id = "watchexec"
+    name = "Watchexec"
+    purl = "pkg:generic/watchexec@1.23.0?arch=arm64"
+    source = "https://github.com/watchexec/watchexec/archive/refs/tags/v1.23.0.tar.gz"
+    source-sha256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+    stacks = ["*"]
+    uri = "https://github.com/watchexec/watchexec/releases/download/v1.23.0/watchexec-1.23.0-aarch64-unknown-linux-musl.tar.xz"
+    sha256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```
Comment (anthonydahanne) [inline buildpack.toml:59]: it seems like this `sha` is wrong
see https://github.com/paketo-buildpacks/watchexec/actions/runs/6588485730/job/17900838767?pr=168

```
  Caching Watchexec 1.23.0
    Downloading from https://github.com/watchexec/watchexec/releases/download/v1.23.0/watchexec-1.23.0-aarch64-unknown-linux-musl.tar.xz
    Verifying checksum
unable to download https://github.com/watchexec/watchexec/releases/download/v1.23.0/watchexec-1.23.0-aarch64-unknown-linux-musl.tar.xz
sha256 for /home/runner/carton-cache/e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/watchexec-1.23.0-aarch64-unknown-linux-musl.tar.xz 3a977fa419d7dafe07b119a9c0b3c53311b0b624a388647ce0f0e12b254f4351 does not match expected e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
```

**Comment by anthonydahanne** [comment]
almost there! after I update `libpak` we can come back and merge!

File: buildpack.toml (hunk 7 above)
Comment (dmikusa) [inline buildpack.toml:59]:
```suggestion
    sha256 = "3a977fa419d7dafe07b119a9c0b3c53311b0b624a388647ce0f0e12b254f4351"
```

File: buildpack.toml [hunk 8]
```diff
@@ -32,26 +32,36 @@ api = "0.7"
...
 
-[[stacks]]
-  id = "io.buildpacks.stacks.bionic"
+  [[metadata.dependencies]]
+    cpes = ["cpe:2.3:a:watchexec:watchexec:1.23.0:*:*:*:*:*:*:*"]
+    id = "watchexec"
+    name = "Watchexec"
+    purl = "pkg:generic/watchexec@1.23.0?arch=arm64"
+    source = "https://github.com/watchexec/watchexec/archive/refs/tags/v1.23.0.tar.gz"
+    source-sha256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```
Comment (dmikusa) [inline buildpack.toml:56]:
```suggestion
    source-sha256 = "2a321962669979feef44ea7a6220819d5c916ca939eba41b033ea346a44caa90"
```

File: buildpack.toml [hunk 9]
```diff
@@ -32,26 +32,36 @@ api = "0.7"
...
-    purl = "pkg:generic/watchexec@1.20.5?arch=amd64"
-    sha256 = "33e091870736833b77b103a8e7268c0ae66d756b8d71203405353eb9d40774f2"
-    source = "https://github.com/watchexec/watchexec/archive/refs/tags/cli-v1.20.5.tar.gz"
-    source-sha256 = "2bc04c7ecf58d34a48c3eeea54a76b7e621717cb93305497bea2f6399dd119c6"
-    stacks = ["io.buildpacks.stacks.bionic", "io.paketo.stacks.tiny", "*"]
-    uri = "https://github.com/watchexec/watchexec/releases/download/cli-v1.20.5/watchexec-1.20.5-x86_64-unknown-linux-musl.tar.xz"
-    version = "1.20.5"
+    purl = "pkg:generic/watchexec@1.23.0?arch=amd64"
+    source = "https://github.com/watchexec/watchexec/archive/refs/tags/v1.23.0.tar.gz"
+    source-sha256 = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
```
Comment (dmikusa) [inline buildpack.toml:40]:
```suggestion
    source-sha256 = "2a321962669979feef44ea7a6220819d5c916ca939eba41b033ea346a44caa90"
```

**Comment by dmikusa** [comment]
Does my sha256 for the source look right to you? I grabbed these yesterday and must of messed something up cause they were all wrong today. I think I have them all corrected.

File: buildpack.toml [hunk 10]
```diff
@@ -32,26 +32,36 @@ api = "0.7"
...
-    source = "https://github.com/watchexec/watchexec/archive/refs/tags/cli-v1.20.5.tar.gz"
-    source-sha256 = "2bc04c7ecf58d34a48c3eeea54a76b7e621717cb93305497bea2f6399dd119c6"
-    stacks = ["io.buildpacks.stacks.bionic", "io.paketo.stacks.tiny", "*"]
-    uri = "https://github.com/watchexec/watchexec/releases/download/cli-v1.20.5/watchexec-1.20.5-x86_64-unknown-linux-musl.tar.xz"
-    version = "1.20.5"
+    purl = "pkg:generic/watchexec@1.23.0?arch=amd64"
+    source = "https://github.com/watchexec/watchexec/archive/refs/tags/v1.23.0.tar.gz"
+    source-sha256 = "2a321962669979feef44ea7a6220819d5c916ca939eba41b033ea346a44caa90"
+    stacks = ["*"]
+    sha256 = "9cab0c339ad5f6da34ce6ce00d27dc02e9ba3f5418f81c728b83c1b7feef051d"
```
Comment (anthonydahanne) [inline buildpack.toml:42]: checked, this is the right sha

File: buildpack.toml [hunk 11]
```diff
@@ -32,26 +32,36 @@ api = "0.7"
...
+  [[metadata.dependencies]]
+    cpes = ["cpe:2.3:a:watchexec:watchexec:1.23.0:*:*:*:*:*:*:*"]
+    id = "watchexec"
+    name = "Watchexec"
+    purl = "pkg:generic/watchexec@1.23.0?arch=arm64"
+    source = "https://github.com/watchexec/watchexec/archive/refs/tags/v1.23.0.tar.gz"
+    source-sha256 = "2a321962669979feef44ea7a6220819d5c916ca939eba41b033ea346a44caa90"
+    stacks = ["*"]
+    uri = "https://github.com/watchexec/watchexec/releases/download/v1.23.0/watchexec-1.23.0-aarch64-unknown-linux-musl.tar.xz"
+    sha256 = "3a977fa419d7dafe07b119a9c0b3c53311b0b624a388647ce0f0e12b254f4351"
```
Comment (anthonydahanne) [inline buildpack.toml:59]: checked, this is the right sha

**Comment by anthonydahanne** [comment]
yep, all good to go @dmikusa 
Unfortunately, I can't merge in this project.
@pivotal-david-osullivan or @ForestEckhardt could you review and merge please?

**Review by anthonydahanne — APPROVED** [review]

**Pull request closed without merging**